snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/core/ql/DefaultQlParser.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2018 B2i Healthcare Pte Ltd, http://b2i.sg
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.b2international.snowowl.snomed.core.ql;
+
+import static com.google.common.collect.Maps.newHashMap;
+
+import java.io.StringReader;
+import java.util.List;
+import java.util.Map;
+
+import org.eclipse.emf.ecore.resource.Resource;
+import org.eclipse.emf.ecore.resource.impl.ResourceImpl;
+import org.eclipse.xtext.diagnostics.Severity;
+import org.eclipse.xtext.nodemodel.INode;
+import org.eclipse.xtext.nodemodel.SyntaxErrorMessage;
+import org.eclipse.xtext.parser.IParseResult;
+import org.eclipse.xtext.parser.IParser;
+import org.eclipse.xtext.util.CancelIndicator;
+import org.eclipse.xtext.validation.CheckMode;
+import org.eclipse.xtext.validation.IResourceValidator;
+import org.eclipse.xtext.validation.Issue;
+
+import com.b2international.commons.Pair;
+import com.b2international.commons.StringUtils;
+import com.b2international.snowowl.core.exceptions.BadRequestException;
+import com.b2international.snowowl.core.exceptions.SyntaxException;
+import com.b2international.snowowl.snomed.ql.Disjunction;
+import com.b2international.snowowl.snomed.ql.Query;
+
+/**
+ * @since 6.12
+ */
+public class DefaultQlParser implements QlParser {
```

Comment:
I suggest renaming the Query Language services, APIs to use `SnomedQuery` instead of just `Ql`.
Couple of examples:
- `QlParser` -> `SnomedQueryParser`
- `DefaultQlParser` -> `DefaultSnomedQueryParser`
- `QlExpression` -> `SnomedQueryExpression`